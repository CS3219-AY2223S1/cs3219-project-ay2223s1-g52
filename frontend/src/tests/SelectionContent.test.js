import renderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";
import { render, screen } from "@testing-library/react";
import SelectionContent from "views/selection-content";

it("renders correctly", async () => {
  const tree = renderer.create(<SelectionContent />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders text in cards correctly", async () => {
  render(<SelectionContent />);
  expect(screen.getByText("Easy")).toBeInTheDocument();
  expect(screen.getByText("Medium")).toBeInTheDocument();
  expect(screen.getByText("Hard")).toBeInTheDocument();
});

test("renders easy dialog correctly", async () => {
  render(<SelectionContent />);
  const difficulty = "Easy";
  await userEvent.click(screen.getByText(difficulty));

  expect(
    screen.getByText(`Find a match for Difficulty: ${difficulty}`),
  ).toBeInTheDocument();
  expect(screen.getByText("Match")).toBeInTheDocument();
});

test("renders medium dialog correctly", async () => {
  render(<SelectionContent />);
  const difficulty = "Medium";
  await userEvent.click(screen.getByText(difficulty));

  expect(
    screen.getByText(`Find a match for Difficulty: ${difficulty}`),
  ).toBeInTheDocument();
  expect(screen.getByText("Match")).toBeInTheDocument();
});

test("renders hard dialog correctly", async () => {
  render(<SelectionContent />);
  const difficulty = "Hard";
  await userEvent.click(screen.getByText(difficulty));

  expect(
    screen.getByText(`Find a match for Difficulty: ${difficulty}`),
  ).toBeInTheDocument();
  expect(screen.getByText("Match")).toBeInTheDocument();
});

test("starts matching correctly for easy", async () => {
  render(<SelectionContent />);
  const difficulty = "Easy";
  await userEvent.click(screen.getByText(difficulty));
  await userEvent.click(screen.getByText("Match"));

  expect(
    screen.getByText("Once matched, you will be redirected to a room."),
  ).toBeInTheDocument();
});

test("starts matching correctly for medium", async () => {
  render(<SelectionContent />);
  const difficulty = "Medium";
  await userEvent.click(screen.getByText(difficulty));
  await userEvent.click(screen.getByText("Match"));

  expect(
    screen.getByText("Once matched, you will be redirected to a room."),
  ).toBeInTheDocument();
});

test("starts matching correctly for hard", async () => {
  render(<SelectionContent />);
  const difficulty = "Hard";
  await userEvent.click(screen.getByText(difficulty));
  await userEvent.click(screen.getByText("Match"));

  expect(
    screen.getByText("Once matched, you will be redirected to a room."),
  ).toBeInTheDocument();
});
