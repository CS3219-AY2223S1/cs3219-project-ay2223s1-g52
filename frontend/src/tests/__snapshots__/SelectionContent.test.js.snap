// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiBox-root css-acvaxx"
>
  <h5
    className="MuiTypography-root MuiTypography-h5 css-zmqfxo-MuiTypography-root"
  >
    Select a difficulty to start matching with other users
  </h5>
  <main
    className="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 css-1ld3b9g-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-11l5t4l-MuiGrid-root"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-cfdeft-MuiPaper-root-MuiCard-root"
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <div
              className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <div
                className="MuiBox-root css-1aegnta"
              >
                <span
                  className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
                >
                  Easy
                </span>
              </div>
            </div>
            <span
              className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
            />
          </button>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-11l5t4l-MuiGrid-root"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-cfdeft-MuiPaper-root-MuiCard-root"
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <div
              className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <div
                className="MuiBox-root css-1aegnta"
              >
                <span
                  className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
                >
                  Medium
                </span>
              </div>
            </div>
            <span
              className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
            />
          </button>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-11l5t4l-MuiGrid-root"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-cfdeft-MuiPaper-root-MuiCard-root"
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <button
            className="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <div
              className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <div
                className="MuiBox-root css-1aegnta"
              >
                <span
                  className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
                >
                  Hard
                </span>
              </div>
            </div>
            <span
              className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
            />
          </button>
        </div>
      </div>
    </div>
  </main>
</div>
`;
